src/commands/apextests/list.ts

```diff
@@ -59,22 +57,25 @@ export default class ApextestsList extends SfCommand<ApextestsListResult> {
     const files = readDir.filter((file) => file.endsWith('.cls') || file.endsWith('.trigger'));
     const testMethodsNames: string[] = [];
 
-    // read each file and check for the test methods at the top
-    files.map((file) => {
-      const data = fs.readFileSync(`${directory}/${file}`, 'utf8');
-
-      // try to find, with a RegEx, the test methods listed at the top of the
-      // file with @Tests or @TestSuites
-      const testMethods = data.match(/(@(Tests|TestSuites)).+/g);
-      // for each entry, parse the names
-      // const testMethodsNames = testMethods ? ApextestsList.parseTestsNames(testMethods.join(',')) : [];
-      testMethodsNames.push(...(testMethods ? ApextestsList.parseTestsNames(testMethods.join(',')) : []));
+    const readPromises = files.map(async (file) => {
```

Comment:
**suggestion:** you could do `Promise.all` here

And also, map is used to return element
It is based to use it to do so and avoid mutating external element (like the testMethodsNames array)
As each iteration returns an array you could use flatMap instead of map

```suggestion
    const testMethodsNames = Promise.all(files.flatMap(async (file) => {
```